superset/models/helpers.py

```diff
@@ -1269,17 +1269,12 @@ def _get_top_groups(
 
         return or_(*groups)
 
-    def dttm_sql_literal(
-        self,
-        col: "TableColumn",
-        dttm: sa.DateTime,
-        col_type: Optional[str],
-    ) -> str:
+    def dttm_sql_literal(self, dttm: datetime, col: "TableColumn") -> str:
         """Convert datetime object to a SQL expression string"""
 
         sql = (
-            self.db_engine_spec.convert_dttm(col_type, dttm, db_extra=None)
-            if col_type
+            self.db_engine_spec.convert_dttm(col.type, dttm, db_extra=col.db_extra)
```

Comment:
In the `TableColumn.dttm_sql_literal` method this the `db_extra` is set. I gather it might have been set to `None` previously for connivence/ease.